### PR TITLE
Avoid read ahead for non-aggregating column families

### PR DESCRIPTION
**Pull request by matthpeterson**
PropogatingIterator makes an extra next() call, even for keys that do not require aggregation.  This change delays the call to next() and in some use cases prevents the need to call next() at all.  This can save a substantial amount of time for data that is being skipped by a filter.

### CODE REVIEW

**Comment by bbux-atg** [comment]
I would feel better if there were some unit tests to exercise the changes

File: warehouse/ingest-core/src/main/java/datawave/iterators/PropogatingIterator.java [hunk 1]
```diff
@@ -171,10 +157,6 @@ private boolean aggregateRowColumn() throws IOException {
                     log.trace("Not propogating " + workKey);
                 return false;
             }
-        } else {
-            
-            iterator.next();
```
[the remark this answers is not in the document]
Comment (matthpeterson) [reply warehouse/ingest-core/src/main/java/datawave/iterators/PropogatingIterator.java]: I think the newly added unit tests address this concern.

**Comment by matthpeterson** [comment]
Notes on PropogatingIteratorTest: I didn't remove any actual tests, only unused code.  There were a few duplicate tests, some claiming to test deepCopy but not actually calling deepCopy.  Those were updated to call deepCopy although they're still of questionable value.  I extracted some common code to a method and some constants.  Otherwise, the tests were left unchanged in this class.  

I added a new test class, PropogatingIteratorSeekTest, because these new tests take a different approach.

File: warehouse/ingest-core/src/main/java/datawave/iterators/PropogatingIterator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivakegg) [reply warehouse/ingest-core/src/main/java/datawave/iterators/PropogatingIterator.java]: ah, I see where I was mistaken... thanks

**Pull request merged**